main/src/main/scala/sbt/internal/DefaultBackgroundJobService.scala

```diff
@@ -306,6 +305,28 @@ private[sbt] abstract class AbstractBackgroundJobService extends BackgroundJobSe
       converter: FileConverter,
   ): Classpath =
     copyClasspath(products, full, workingDirectory, hashFileContents = true, converter)
+
+  private[sbt] def pauseChannelDuringJob(state: State, handle: JobHandle): Unit =
+    currentChannel(state) match
+      case Some(channel) =>
+        val level = channel.logLevel
+        channel.setLevel(Level.Error)
+        channel.pause()
+        handle match
+          case t: ThreadJobHandle =>
+            t.job.onStop: () =>
+              channel.setLevel(level)
+              channel.resume()
+              channel.prompt(ConsolePromptEvent(state))
+          case _ => ()
+      case _ => ()
+
+  private[sbt] def currentChannel(state: State): Option[CommandChannel] =
+    state.currentCommand match
+      case Some(e: Exec) if e.source.isDefined =>
+        val source = e.source.get
+        StandardMain.exchange.channelForName(source.channelName)
+      case _ => None
```

Comment:
NIT

```suggestion
    state.currentCommand.flatMap(_.source)
      .map(source => StandardMain.exchange.channelForName(source.channelName))
```

Reply:
That creates a few lambdas whereas the pattern matching compiles into if statements.

Reply:
How bad is that?

Reply:
It kind of depends on case-by-case, but on a hot path it does matter, so I tend to avoid `map(...).getOrElse(y)` style whenever I can just pattern match instead.